content/blog/open-sourcing-workshops-on-demand-part3-understanding-the-backend.md

```diff
@@ -0,0 +1,340 @@
+export WODTYPE=backend
+
+# This main dir is computed and is the backend main dir
+export WODBEDIR=/home/wodadmin/wod-backend
+
+# BACKEND PART
+# The backend dir has some fixed subdirs
+# wod-backend (WODBEDIR)
+#    |---------- ansible (ANSIBLEDIR)
+#    |---------- scripts (SCRIPTDIR defined in all.yml not here to allow overloading)
+#    |---------- sys (SYSDIR)
+#    |---------- install
+#    |---------- conf
+#    |---------- skel
+#
+export ANSIBLEDIR=$WODBEDIR/ansible
+export SYSDIR=$WODBEDIR/sys
+
+# PRIVATE PART
+# These 3 dirs have fixed names by default that you can change in this file
+# they are placed as sister dirs wrt WODBEDIR
+# This is the predefined structure for a private repo
+# wod-private (WODPRIVDIR)
+#    |---------- ansible (ANSIBLEPRIVDIR)
+#    |---------- notebooks (WODPRIVNOBO)
+#    |---------- scripts (SCRIPTPRIVDIR)
+#
+PWODBEDIR=`dirname $WODBEDIR`
+export WODPRIVDIR=$PWODBEDIR/wod-private
+export ANSIBLEPRIVDIR=$WODPRIVDIR/ansible
+export SCRIPTPRIVDIR=$WODPRIVDIR/scripts
+export SYSPRIVDIR=$WODPRIVDIR/sys
+export WODPRIVNOBO=$WODPRIVDIR/notebooks
+WODPRIVINV=""
+# Manages private inventory if any
+if [ -f $WODPRIVDIR/ansible/inventory ]; then
+        WODPRIVINV="-i $WODPRIVDIR/ansible/inventory"
+        export WODPRIVINV
+fi
+
+# AIP-DB PART
+export WODAPIDBDIR=$PWODBEDIR/wod-api-db
+
+# FRONTEND PART
+export WODFEDIR=$PWODBEDIR/wod-frontend
+
+# These dirs are also fixed by default and can be changed as needed
+export WODNOBO=$PWODBEDIR/wod-notebooks
+export STUDDIR=/student
+#
+export ANSPLAYOPT="-e PBKDIR=staging -e WODUSER=wodadmin -e WODBEDIR=/home/wodadmin/wod-backend -e WODNOBO=/home/wodadmin/wod-notebooks -e WODPRIVNOBO=/home/wodadmin/wod-private/notebooks -e WODPRIVDIR=/home/wodadmin/wod-private -e WODAPIDBDIR=/home/wodadmin/wod-api-db -e WODFEDIR=/home/wodadmin/wod-frontend -e STUDDIR=/student -e ANSIBLEDIR=/home/wodadmin/wod-backend/ansible -e ANSIBLEPRIVDIR=/home/wodadmin/wod-private/ansible -e SCRIPTPRIVDIR=/home/wodadmin/wod-private/scripts -e SYSDIR=/home/wodadmin/wod-backend/sys -e SYSPRIVDIR=/home/wodadmin/wod-private/sys"
+export ANSPRIVOPT=" -e @/home/wodadmin/wod-private/ansible/group_vars/all.yml -e @/home/wodadmin/wod-private/ansible/group_vars/staging"
+```
+
+`r﻿andom.sh` ﻿exports the randomly generated password.
+
+`f﻿unctions.sh` is a library of shell functions used by many scripts among which `procmail-action.sh`.  We will see the details shortly below.
+
+4- `procmail-action.sh` c﻿alls the necessary functions and scripts to perform the  **CREATE** operation.
+
+5- `get_session_token()` This function retrieves the necessary token to make api call to the api-db server.
+
+6- `get_workshop_name()`  This function extracts the workshop name from the mail body. In Body: One will find the workshop name: for example, **WKSHP-API101** 
+
+7- `get_workshop_id()` Out of the workshop name, the function `get_workshop_id()` will get the workshop's id from the api-db server.
```

Comment:
Instead of "Out of the workshop name, the...", I think you mean "From the workshop name, the..."

Be consistent if you decide to capitalize ID and API-DB server per my earlier recommendation.

Reply:
OK for ID but not for api-db server, this is a name as for backend, frontend, they represent types of servers...